concrete-core-fixture/src/fixture/lwe_ciphertext_cleartext_discarding_multiplication.rs

```diff
@@ -0,0 +1,179 @@
+use concrete_commons::numeric::UnsignedInteger;
+use concrete_commons::parameters::LweDimension;
+use concrete_core::prelude::{
+    CleartextEntity, LweCiphertextCleartextDiscardingMultiplicationEngine, LweCiphertextEntity,
+};
+
+/// A fixture for the types implementing the `LweCiphertextCleartextDiscardingMultiplicationEngine`
+/// trait.
+pub struct LweCiphertextCleartextDiscardingMultiplicationFixture;
+
+#[derive(Debug)]
+pub struct LweCiphertextCleartextDiscardingMultiplicationParameters {
+    pub noise: Variance,
+    pub lwe_dimension: LweDimension,
+}
+
+#[allow(clippy::type_complexity)]
+impl<Precision, Engine, InputCiphertext, Cleartext, OutputCiphertext>
+    Fixture<Precision, Engine, (InputCiphertext, Cleartext, OutputCiphertext)>
+    for LweCiphertextCleartextDiscardingMultiplicationFixture
+where
+    Precision: IntegerPrecision,
+    Engine: LweCiphertextCleartextDiscardingMultiplicationEngine<
+        InputCiphertext,
+        Cleartext,
+        OutputCiphertext,
+    >,
+    InputCiphertext: LweCiphertextEntity,
+    Cleartext: CleartextEntity,
+    OutputCiphertext: LweCiphertextEntity<KeyDistribution = InputCiphertext::KeyDistribution>,
+    Maker: SynthesizesCleartext<Precision, Cleartext>
+        + SynthesizesLweCiphertext<Precision, InputCiphertext>
+        + SynthesizesLweCiphertext<Precision, OutputCiphertext>,
+{
+    type Parameters = LweCiphertextCleartextDiscardingMultiplicationParameters;
+    type RepetitionPrototypes = (
+        <Maker as PrototypesLweSecretKey<Precision, InputCiphertext::KeyDistribution>>::LweSecretKeyProto,
+        <Maker as PrototypesCleartext<Precision>>::CleartextProto,
+    );
+    type SamplePrototypes = (
+        <Maker as PrototypesPlaintext<Precision>>::PlaintextProto,
+        <Maker as PrototypesLweCiphertext<Precision, InputCiphertext::KeyDistribution>>::LweCiphertextProto,
+        <Maker as PrototypesLweCiphertext<Precision, InputCiphertext::KeyDistribution>>::LweCiphertextProto,
+    );
+    type PreExecutionContext = (InputCiphertext, Cleartext, OutputCiphertext);
+    type PostExecutionContext = (InputCiphertext, Cleartext, OutputCiphertext);
+    type Criteria = (Variance,);
+    type Outcome = (Precision::Raw, Precision::Raw);
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![LweCiphertextCleartextDiscardingMultiplicationParameters {
+                noise: Variance(LogStandardDev::from_log_standard_dev(-15.).get_variance()),
+                lwe_dimension: LweDimension(600),
+            }]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+        let proto_secret_key = maker.new_lwe_secret_key(parameters.lwe_dimension);
+        let raw_cleartext = Precision::Raw::uniform_zero_centered(1024);
```

Comment:
Why this value 1024? It seemed to me in the original test we were picking a value between 0 and 1024-512 = 512, but I probably don't understand the original test :sweat_smile: .